syntax = "proto3";

service PipelineServer {
    // Pipeline sends processed frames + annotations
    rpc SendProcessedData (stream ProcessedFrameRequest) returns (UploadResponse);

    // Training machines request processed batches
    rpc StreamProcessedBatches (BatchRequest) returns (stream ProcessedBatch);
}

message ProcessedFrameRequest {
    string video_name = 1;
    string frame_name = 2;
    bytes frame_data = 3;
    string annotation_json = 4;
}

message UploadResponse {
    string message = 1;
}

message BatchRequest {
    int32 batch_size = 1;
}

message ProcessedBatch {
    repeated bytes frames = 1;
    repeated string annotations = 2;
}
